### PR TITLE
Update c-blosc to v1.21.1

### PR DESCRIPTION
**Pull request by avalentino**
In order to merge a small modification was required to the `subtree-merge-blosc.sh` script:

```diff
$ git diff master subtree-merge-blosc.sh
diff --git a/subtree-merge-blosc.sh b/subtree-merge-blosc.sh
index 83327018..9d13f789 100755
--- a/subtree-merge-blosc.sh
+++ b/subtree-merge-blosc.sh
@@ -68,7 +68,7 @@ git fetch $remote $blosc_tag_long || exit 1
 # subtree merge it
 #git merge --squash -s subtree FETCH_HEAD || exit 1
 # if this doesn't work you could use:
- git merge --squash -Xsubtree=c-blosc -Xtheirs FETCH_HEAD || exit 1
+ git merge --allow-unrelated-histories --squash -Xsubtree=c-blosc -Xtheirs FETCH_HEAD || exit 1
 if git diff --staged --quiet ; then
     fatal "nothing new to be committed"
 else
```

@FrancescAlted should I file a PR http://github.com/Blosc/subtree-merge-blosc or it is better to move to `git submodule`?

### CODE REVIEW

**Comment by FrancescAlted** [comment]
IMO it would be better to change to submodules, but you have the last say.

**Comment by avalentino** [comment]
@FrancescAlted OK I will change it before merging.
Do you have an idea why appveyor keeps to be triggered?
I have removed all the appveryor related files form the repo and rebased this PR.
I would expect that now appveyor runs are no longer started but this is not the case.
Unfortunately I do not have access to the PyTables project on Appveyor.com.

**Comment by FrancescAlted** [comment]
Ha, that's interesting.  No idea on why the AppVeyor CI is triggering after your cleanup.  Unfortunately, I do not have access to the PyTables project on Appveyor.com.  Maybe @tomkooij has a hint on this?

**Comment by avalentino** [comment]
@FrancescAlted now it is ready for review.
I have also removed webhooks for appveyor for the PyTables repository settings, hoping that the CI on appveyor stops to be triggered.

File: RELEASE_NOTES.txt [hunk 1]
```diff
@@ -45,6 +45,7 @@ Improvements
   that the version to which the program was linked to. This problem is known as "DLL Hell".
   With the renaming of the HDF5 DLL to `pytables_hdf5.dll` these problems should be solved.
 - Stop using ppveyor and deprecated ci-helpers (closes :issue:`827`).
+- Switch to `git submodule` for teh management of vendored c-blosc sources.
```
Comment (FrancescAlted) [inline RELEASE_NOTES.txt:48]: teh -> the

**Review by FrancescAlted — APPROVED** [review]
Other than a small typo, this looks good to me.

**Comment by avalentino** [comment]
Thanks @FrancescAlted

**Pull request merged**